### PR TITLE
hotfix(core) bring back x-forwarded-host

### PR DESCRIPTION
**Pull request by thefosk**
Brings back the `X-Forwarded-Host`.

### CODE REVIEW

**Comment by p0pr0ck5** [comment]
Odd, do we know why we have these failing tests?

**Comment by p0pr0ck5** [comment]
@thefosk can you rebase against master? seems the test failures are unrelated, but want to make sure this change doesnt break anything unnoticed.

**Comment by thefosk** [comment]
@p0pr0ck5 @thibaultcha do we agree on this approach? Seems like there are other solutions on the table.

**Comment by p0pr0ck5** [comment]
IMO this seems reasonable, this is what we will do in 0.11 if the client is untrusted, or X-F-H isn't set.

**Comment by thibaultcha** [comment]
Not opposed to it. Maybe some tests?

**Comment by thefosk** [comment]
@thibaultcha added tests

File: kong/templates/nginx_kong.lua [hunk 1]
```diff
@@ -121,6 +121,7 @@ server {
 
         proxy_http_version 1.1;
         proxy_set_header X-Real-IP $remote_addr;
+        proxy_set_header X-Forwarded-Host $host;
```
Comment (kikito) [inline kong/templates/nginx_kong.lua]: On the most recent master, there is already a header called `X-Forwarded-Host`: https://github.com/Mashape/kong/blob/master/kong/templates/nginx_kong.lua#L126

File: spec/02-integration/05-proxy/12-upstream_headers.lua [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  local client
+  setup(function()
+    assert(helpers.dao.apis:insert {
+      name = "mockbin",
+      hosts = { "test.com" },
+      strip_uri = true,
+      upstream_url = "http://mockbin.com"
+    })
+
+    assert(helpers.start_kong())
```
Comment (kikito) [inline spec/02-integration/05-proxy/12-upstream_headers.lua]: We now have a way to test without using mockbin. This needs to be changed to 
```
    assert(helpers.start_kong({
      nginx_conf = "spec/fixtures/custom_nginx.template",
    }))
```

File: spec/02-integration/05-proxy/12-upstream_headers.lua [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+local cjson = require "cjson"
+
+describe("Upstream Headers", function()
+  local client
+  setup(function()
+    assert(helpers.dao.apis:insert {
+      name = "mockbin",
+      hosts = { "test.com" },
+      strip_uri = true,
+      upstream_url = "http://mockbin.com"
```
Comment (kikito) [inline spec/02-integration/05-proxy/12-upstream_headers.lua]: We now have a way to test without using mockbin. This needs to be changed to
```
      upstream_url = helpers.mock_upstream_url,
```

**Comment by thibaultcha** [comment]
This did not make it to 0.10.4 - closing now.

**Pull request closed without merging**